cpp/include/raft/spatial/knn/detail/ivf_pq_build.cuh

```diff
@@ -1365,7 +1367,12 @@ auto build(raft::device_resources const& handle,
   }
 }
 
-static const int serialization_version = 1;
+// Serialization version 2
+// No backward compatibility yet; that is, can't add additional fields without breaking
+// backward compatibility.
+// TODO(hcho3) Implement next-gen serializer for IVF that allows for expansion in a backward
+//             compatible fashion.
+static const int serialization_version = 2;
 
```

Comment:
We've already had a problem with ivf-pq once, that `index` members changed but we forgot to update the load/save functions. Would be nice to detect these changes when updating the code, but I'm not aware of any good way to do this in C++.
As a very crude approximation, what do you think about adding something like this to `load` and `save` functions?
```C++
static_assert(sizeof(ivf_pq::index<uint64_t>) == {literal}, "The size of the index struct has changed since the last update; paste in the new size and consider updating the save/load logic");
```

Reply:
Done